### ISSUE TITLE
[BUG] Erroneous ILS approach: at finale direct to first waypoint of FP

### ISSUE THREAD

**Issue by JLD95** [flybywiresim/a32nx#3609]
**Mod Version**

2021-02-18T03:38:56+00:00

**Describe the bug**

At start of finale, FMS jumps to the first waypoint of flight plan

**To Reproduce**

1.  I use this flight plan:  EGPE/23 ATF RIVOT UPGAS RIBSO RIBS1D EDDH/15
2. At DH456 waypoint, FMS make a direct to ATF waypoint the first waypoint of flight plan
3. 

**Expected behavior**

a correct ILS approach

**Actual behavior**

![2021-02-18 (4)](https://user-images.githubusercontent.com/5894146/108401331-858a0d80-721c-11eb-8725-07d211351aba.jpg)

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

**Comment by Drachondius** [comment]
Hey, first comment!
First: Thank you all for your great work!
Second: +1 for this issue. Made an update yesterday (development build). Did 3 flights to different locations (EDDM/EDDK/EDDL) At first point of STAR Airbus banks full left or right and targets first waypoint in flightplan. Correction via DCT only leads to DCT point but doesnt pic up flight plan again.

**Comment by krieke58** [comment]
Hi,
same here yesterday. Flight EGKK - LEBL. On entering STAR aircraft tracks back to XANAX (first waypoint EGKK). Corrected via DIR TO, works but after this point goes again to XANAX, corrected again and intercepted the ILS correctly. Landing fine.

**Comment by d3cadeaf** [comment]
Got the same problem yesterday at LOWW approach. Hope it will be fixed soon.

**Comment by pareil6** [comment]
If possible, could someone report the exact route + approach they were flying when this bug occurred? I've seen it before but haven't been able to consistently reproduce it, which makes debugging it a pain. I _think_ it's caused by a missing flight plan version update around the time we reach the approach - but haven't been able to test that properly.

**Comment by JLD95** [comment]
See my first post:. EGPE/23 ATF RIVOT UPGAS RIBSO RIBS1D EDDH/15. Ils15

**Comment by Drachondius** [comment]
Update: updated to latest dev build. Flight to LKPR Prague 24 ILS. No problems, no need for directs.

**Comment by DFWSupertrooper** [comment]
I've flown 5 good flights but just happened to me EGKK-LSZH after last update FBW Custom (8a1c3b3).  Route was EGCC 26L BOGN1X BOGNA WAFFU ELDAX LESDO RANUX NEBAX BLM BLM2G.  The FPL included all of the correct SID/STAR/Transition points according to EuroCharts.  

Turn to first departure waypoint KK##? (I forgot the number) happened after passing GIPOL on descent at FL070.  

Switched to HDG mode until capture of ILS.  Landed without further incident.

**Comment by DFWSupertrooper** [comment]
@pareil6 OK, 2nd flight in a row, using newest update from a couple of hours ago (FBW custom):  EGCC 23L SANB1Y SANBA TNT SAPCO DTY FINMA SILVA BENSU WOD MID DRAKE SITET ETRAT DVL LGL BOBSA BOBS9W RNAV20 LFPO.  Total disaster.  

On descent, immediately after ATC cleared to ODILO (final waypoint in BOBS9W STAR), on or about reaching SOTIP, A/C began to turn towards D0 (initial false waypoint at departure).  I engaged DIR TO VASOL (a waypoint that was not in my original FPL but was injected by MSFS or the A/C into the MCDU at origin).  A/C followed and then upon reaching VASOL, turned towards D0 again.  I engaged DIR TO PO615 (listed as next in the FPL after DECEL).  A/C engaged that course and upon reaching turned back towards VASOL.  I engaged HDG mode, flew to PO420 (initial waypoint in RNAV approach).  A/C flew and then tried to go back to VASOL again after passing.  Used HDG mode to navigate to FP020 (which would be the normal GLS intercept) and engaged APPR close to that waypoint, but on what should have been Final App, A/C then tried to go to BENSU (a waypoint _wayyyyy back_ towards departure??).  I disengaged the A/P and landed manually.

This is a show stopper for RNAVs.  At least on ILS you can get to the LOC capture and engage for landing.  

To recap, it was happening on every flight between around the time the EFB was added (not sure exactly when) up to the WU3 release, at which point all of a sudden everything was great on five straight flights (RNAV & ILS), then as of yesterday, BOOM, back again on every flight so far.

For reference, I update FBW every day with latest FBW Custom if that helps.

**Comment by DFWSupertrooper** [comment]
@pareil6 another thing I've noticed is that when the issue begins, the FPL page in the MCDU returns to original state (at origin).

**Comment by rpoulachon** [comment]
Hi, i have experiment the same issue from gatewick to Rouen in France.I have Lost the last waypoints suddenly just after the decelerrate point.
Best regards

**Comment by Tibu73-ui** [comment]
Same problem here. On aproach to RWY 07L EDDF.
All the STAR waypoints dissapear on ND after performs a DIRECT TO:

**Comment by DFWSupertrooper** [comment]
Using 344301a FBW Custom, just flew a flawless flight from KLAS-KDEN.  Issue did not present.  I haven't yet been able to determine a pattern.  Could it have something to do with certain airports and/or Navigraph?  I only ask because I recall that FSDT KORD had to be placed in a higher priority in the _Content_ file in order to avoid Navigraph causing problems with the airport's built-in nav data.  Totally guessing here...not even sure that built-in airport data affects anything outside of the final approach.

Here's what I know: 

- Every airport I fly into/out of is 3rd Party or Asobo handcrafted (no defaults).
- Issue seems _more_ prevalent in European routes than US-based, but it's _not_ 100% split.

Other than those facts, I'm at a loss.  I'll keep flying and reporting which routes it's happening on and see if I can isolate a pattern.

**Comment by mio9** [comment]
Can confirm this happening in several airports in Asia as well, non-handcrafted, without addon sceneries (i.e. RPLL, RCTP, VHHH)

**Comment by DFWSupertrooper** [comment]
Perfect flight EDDM 08R INPU3E INPUD Y102 ALIBU L604 MAPOX DCT SPY DCT RAVLO Y70 OTBED L60 MAMUL L613 HALIF UN590 APPLE APLE1G ILS23 EGPF.  Issue did not present.

**Comment by DFWSupertrooper** [comment]
Issue reoccurred on next flight: EGNM 32 NELS3W NELSA DCT CROFT UL612 HON UN859 DVL DCT BOBSA BOBS9E ILS02 LFPO.

Once passed W/P ODILO en route to PO623 (MSFS inserted W/P), A/C engaged turn towards BOBSA.  I engaged HDG mode, then DIR TO CI02 to intercept LOC (skipped PO623, DECL, USR).  Intercepted ILS02 without issue and landed safely.

**Comment by DFWSupertrooper** [comment]
Perfect flight KMSY 20 DCT AEX DCT JJANE DCT ZANTO DCT KA33W DCT RGINA DCT ILC DCT INYOE DYAMD5 ILS28R KSFO. Issue did not present.

**Comment by DFWSupertrooper** [comment]
Perfect flight KMKE 01L UECKR6 SAMPL DCT FWA DCT ROD DCT HVQ GIBBZ4 ILS19L KIAD. Issue did not present.

**Comment by DFWSupertrooper** [comment]
Perfect flight KMIA 27 SKIPS2 SKIPS DCT MADIZ DCT BELAC DCT FORST DCT JOSES DCT ETBOD DCT RODBU DCT FRIST DCT BEROX DCT GONNE DCT BEXER WEST4 RNAV10 TNCB.  Issue did not present.

**Comment by DFWSupertrooper** [comment]
Perfect flight KATL 27R VARNM2 GLAZR DCT IIU DCT BONNT VEECK5 ILS28R KORD.  Issue did not present.

**Comment by DFWSupertrooper** [comment]
Perfect flight KLAX 25R SUMMR2 MCKEY DCT LIN DCT LMT DCT HRMNS DCT SUMMA DCT TAOMA DCT PAE GRIZZ6 ILS26L CYVR.  Issue did not present.

**Comment by DFWSupertrooper** [comment]
Perfect flight KIAD 19L MCRAY2 MCRAY  IHD  LEJOY  AVERE  KODIE  KOZAR  LIVER  LEPEW  SRPEN  DJB  RIEKE  WOOST  CRL  ALPHE  HASTE  VIO  AUGER  WILDD  NICKL  ITMUW  OGZEV  GGW  CFCKK  YQL  YXC GLASR1 ILS16L KSEA.  Issue did not present.

**Comment by ebisettia** [comment]
Hello: FP EGGL to EIDW. I encountered 2 issues. The first issue was that a waypoint that wasn't in the flight plan was added, curiously was EIDW after (DECEL) and BAGSO. I wasn't able to delete it, no matter how many times I tried, instead, every time I tried, MCDU deleted the waypoint before EIDW. Then the issue N. 2 was that after KERAV MCDU selected EIDW (video 1) so I had to select the next waypoint manually, but, after the next waypoint it selected BAGSO (video 2).


FP from Navigfraph: EGLL/27R CPT3F CPT UL9 KENET N14 MEDOG L18 ABLIN DCT BAGSO BAGS1K EIDW/KERAV.I16
![Microsoft Flight Simulator Screenshot 2021 02 23 - 18 46 23 85](https://user-images.githubusercontent.com/75552858/109062897-84426000-76b6-11eb-85a5-aed11d7489fe.png)
[EGLLEIDW.zip](https://github.com/flybywiresim/a32nx/files/6038708/EGLLEIDW.zip)
https://drive.google.com/file/d/1sBPxAxEM6zwoOIumkxD0VbsigXemV9Va/view?usp=sharing
https://drive.google.com/file/d/1RBEndZilTY7SGH7_rP7rCaBwle0Jh6Oc/view?usp=sharing

**Comment by DFWSupertrooper** [comment]
@ebisettia that's definitely not right, but I think that's a different issue than the one being catalogued here (at least I think...).  What gets imported into the MCDU from an outside source is not at issue in this thread...these are all instances where the import/load is fine, but the A/C won't follow it on descent/approach all the time.  You amy wanna consider looking thru existing threads that may be more related, or start a new one, especially with the great documentation you've got.  An admin can correct me if that's not right??

**Comment by rpoulachon** [comment]
hi everyone, just to understand. all the issues we encountered on this topic are due to the Asobo flight model or the Flybywire mod ? Do you notice the same issues on the standard A320 delivered by Asobo ? Thank you and sorry for my naive question.

**Comment by ebisettia** [comment]
> @ebisettia that's definitely not right, but I think that's a different issue than the one being cataloged here (at least I think...). What gets imported into the MCDU from an outside source is not at issue in this thread...these are all instances where the import/load is fine, but the A/C won't follow it on descent/approach all the time. You may wanna consider looking thru existing threads that may be more related or start a new one, especially with the great documentation you've got. An admin can correct me if that's not right??

@DFWSupertrooper Thanks, actually I was afraid to open a new issue and breaking any rules by doing that and I thought this Subject was the closest to mine.  Something I forgot to explain is that my flight plan was created and imported from SIMEBRIEF to the MCDU and Navigraph, then completed in the MCDU with the STAR and APP using Navigraph's sequence. So I'm gonna risk it and create a new issue, hopefully, I won't get banned or something, but there is a thing, I don't know how to find the FBW Dev version was on my sim at that time, it was yesterday and today I forgot to look for that and stupid me updated the FBW Dev version.

**Comment by DFWSupertrooper** [comment]
> hi everyone, just to understand. all the issues we encountered on this topic are due to the Asobo flight model or the Flybywire mod ? Do you notice the same issues on the standard A320 delivered by Asobo ? Thank you and sorry for my naive question.

@rpoulachon the issue on _this_ thread has to do with the FBW A/P in the FBW Custom version.  They are not related to the default Asobo A320.

@ebisettia, no problem...you won't be banned for opening an issue.  The guys here are very friendly and want to make this A/C great.  They'll let you know if the issue is not related or in the right place.  You'll only get banned if you're a jerk ;).  Haha.

**Comment by ebisettia** [comment]
> > hi everyone, just to understand. all the issues we encountered on this topic are due to the Asobo flight model or the Flybywire mod ? Do you notice the same issues on the standard A320 delivered by Asobo ? Thank you and sorry for my naive question.
> 
> @rpoulachon the issue on _this_ thread has to do with the FBW A/P in the FBW Custom version. They are not related to the default Asobo A320.
> 
> @ebisettia, no problem...you won't be banned for opening an issue. The guys here are very friendly and want to make this A/C great. They'll let you know if the issue is not related or in the right place. You'll only get banned if you're a jerk ;). Haha.

Thanks, I just submitted it. Let's see how it goes. Cheers.

**Comment by DFWSupertrooper** [comment]
Perfect flight AGAIN.  KSEA 16L DCT SEA DCT STEVS DCT ZAXUL DCT LEZLE DCT MUPDE DCT TOR DCT WILOP DCT SGF DCT ZAVEL DCT LUKKY DCT MEM DCT GOOGY DCT OTK PIGLT6 ILS18R KMCO.  Issue did not present.

Tomorrow, I will begin testing again in Europe and see if the issue returns (this makes 7 _straight_ US-based flights without the issue presenting at all).

**Comment by DFWSupertrooper** [comment]
Updated FBW Custom to newest version.  

Perfect flight.  EGCC 23L SANB1Y SANBA DCT SAPCO DCT WELIN DCT LAM DCT ITVIP DCT DVR DCT RINTI DCT OTGUV DCT ELTEG DCT IBRIG DCT OLMOL DCT LESDO DCT RANUX DCT ARPUS ARPU8K ILS33 LFSB.  Issue did not present.

Note:  I engaged HDG mode prior to IAP, not b/c of an error but to fly the appropriate approach pattern after approach clearance.

**Comment by DFWSupertrooper** [comment]
Perfect flight. LIML 18 CANNE8C CANNE DCT BASGO DCT DITON DCT ELVET DCT SARRI DCT TOMPI DCT MAPIG DCT TOLEY DCT IDARO DCT RUDOT DCT BIKMU BIKM1X ILS05R EDDL. Issue did not present.

**Comment by DFWSupertrooper** [comment]
Another perfect flight.  EDDM 08R INPU3E INPUD DCT ERMEL DCT SPESA DCT BUB DCT DENUT DCT COA DCT KEGIT DCT SASKI DCT INLOD DCT SUMUM DCT LOGAN DCT CLN DCT TNT DCT LISTO DCT VABKA DCT WAL DCT MALUD DCT DONAX DCT BAGSO BAGS2L ILS28L EIDW.  Issue did not present.

OK.  I give up for now.  Maybe this got fixed somehow and no one knows??

**Comment by luiscurtolo** [comment]
Had this problem yesterday and today between EHAM-EGLC. 

The bank starts briefly before reaching the first WP of the STAR and makes my PILOT PFD freeze.

**Comment by DFWSupertrooper** [comment]
@luiscurtolo do you have the exact route you used?  I'd like to try it with the same parameters and see if I also get that issue.  I've never seen the PFD _freeze_ as you describe in conjunction with this issue.  You may be experiencing something different.

**Comment by luiscurtolo** [comment]
> @luiscurtolo do you have the exact route you used? I'd like to try it with the same parameters and see if I also get that issue. I've never seen the PFD _freeze_ as you describe in conjunction with this issue. You may be experiencing something different.

The route out Simbrief. (EHAM - EGLC) 

ROUTE ID: DEFRTE
EHAM/27 VOLLA1P IDRID L980 XAMAN XAMAN1C EGLC/27

The issue began after clearance to JACKO (or something like that), after LOGAN. It started a left bank, like usual with the bug. 

I was flying the Dev version 472cf52. 

About the FREEZED PILOT SIDE FPD, its a new bug, theres a post in the issues forum. Seems to affect both dev and custom versions. 

let me know, good luck with the landing haha

**Comment by JLD95** [comment]
Hi,
with the last dev "2021-02-26T11:35:45", I retry the inital flight plan mentionned in my issue: 
route EGPE/23 DCT ATF DCT RIVOT DCT UPGAS DCT RIBSO RIBSO1D EDDH/15

I obtain exactly the same issue: at DH456 waypoint, FMS makes a direct to ATF waypoint, the first waypoint of flight plan

**Comment by DFWSupertrooper** [comment]
@luiscurtolo OK, so the route for the most part worked, but yes, there was a circling at JACKO.  It's strange b/c on the chart, that is actually a hold point, and I was above the prescribed FL (still on my down to where I was instructed by ATC).  I don't know if it automatically put me in a holding pattern there or what (not supposed to but who knows)...I engaged DIR TO the next waypoint in the transition (NONVA) and the A/C continued as expected the remainder of the route.  I intercepted the LOC around LCE07.  the only sticking point was the G/S.  I should have intercepted at LAVNO at FL030, and I didn't - it didn't pick it up until the FAF (by which point I should have been at FL020).  The result was I came in really hot (about 150K, when I should have been at about 134K) and made a hard landing.  I did, however, walk away, although I'm pretty sure some folks will never fly KLM again!

**Comment by luiscurtolo** [comment]
@DFWSupertrooper thanks for testing it. 

I had no problem with the follow through after using the DCT NONVA, started the GS at 3000. 

I think it's more simple than what you said about the holding point cuz, at least for me, I was descending already and the AP never tried to correct anything. 

I believe it's just the commom problem with entering a STAR. 

Hahahaha about the landing, LC is hard. Try coming down with spoilers up. Sometimes I have to have it like full extended to avoid the gain in airspeed. Real life they use spoilers but maybe not that extended. 

Thanks and hope they fixe this bug. 
Cheers

**Comment by DFWSupertrooper** [comment]
So, the issue presented on another flight this morning: flying EGKK 08R SFD9P SFD Y47 DRAKE N859 SITET UN859 LGL A34 BOBSA BOBS9E ILS06 LFPO.  

Immediately after hitting ODILO (final W/P on descent) the A/C turned towards D0 (the initial departure W/P).  This coincidentally occurred at the exact same time I entered the ILS frequency (having forgotten to do so at load but I don't think that's related since that was never part of the equation before).

I DIR TO PO609 (the next W/P) and the A/C returned to correct course thence.  

Unrelated, I was nowhere near the correct FL, and coincidentally never intercepted the G/S.  Bright skies, I took her in manually and landed w/o issue.  Even after manually descending to the correct FL and seeing the diamond for G/S on the line, I couldn't get the A/C to engage it...unrelated but worth mentioning.

About to retry another flight where I had the issue before to see if it happens again at that airport.

**Comment by DFWSupertrooper** [comment]
Happened again going into LSZH (EGGP 27 NANT2V NANTI DCT KEPAD DCT HON DCT DTY DCT WOBUN DCT BUZAD DCT BPK DCT LAM DCT ITVIP DCT DVR DCT RINTI DCT OTGUV DCT ELTEG DCT IBRIG DCT OLMOL DCT LESDO DCT RANUX DCT NEBAX DCT BLM BLM2G ILS14 LSZH).

A/C, after passing GIPOL (final W/P in descent), turned towards D269B, the first false departure W/P out of EGGP.  Redirected to ZH713, and completed flight w/o further incident.

It seems to be these false w/p on approach doing it, and only at certain airports (maybe the ones using them).  What causes them to be created?  Why can't the FPL simply follow the SID, ENR, STAR, Transition, Approach points?

**Comment by JLD95** [comment]
Hi, 
new route with ILS07 approach GCTS don't work in automatic:
LPMA/05 GOSGA1Z GOSGA UQ143 ORTIS ORTIS1X GCTS/07

**Comment by AdamNZ** [comment]
I'll post some screenshots shortly .... but I found the problem isn't so much as the a/p starts a L.H. turn ... it's that  **it's going back to the previous waypoint**!!! If you [patiently] allow it to back-track, it then flies the remainder of the route correctly.

**Comment by luiscurtolo** [comment]
Yes, and for the most part if you identify a USR waypoint jump to the next waypoint using DIRECT. This helps. But it's an annoying bug as it can happen without a USR point.

**Comment by patascii** [comment]
Same thing for me. LFML/13L FJR6S FJR G393 AFRIC AFRIC6L LFBO/14R  when arriving the STAR, A32NX turn left or right to w/p of the SID. No problem with A320 of Asobo. Tested 3 times. It seems to me that this happen after the last patch 1.13.16.

**Comment by mtjones83** [comment]
Hi,
Had the same issue tonight while inbound to EGLL on the LOGA2H arrival via LAM. With LAM as the next waypoint, shortly before reaching LAM the autopilot changed to navigate to the first waypoint in my flight plan (from Gothenburg) and began a right turn towards it.

I resolved by quickly selecting DIR TO LAM; but then under ATC vectors I left LAM on a heading so no evidence to share on what it might have done (or not) once I reached LAM i.e. continue to next waypoint on STAR; or do something completely unexpected.

![image](https://user-images.githubusercontent.com/73524148/109576808-6036b280-7aec-11eb-87f9-f233de57158d.png)

Cheers,
Mike

**Comment by Benjozork** [comment]
To update everyone, this is something that will be permanently resoled with our custom flight plan system and FMGC, currently being worked on. Stay tuned.

**Comment by AdamNZ** [comment]
Now that's great news! Saves us having to keep posting the same bug ... and it means I can now delete the 50 screenshots I took LOL!

**Comment by CORTRacer31** [comment]
This has happened to me twice now over my last two FBW A32NX flights, and I think I might have an idea why it's happening. When building the flight plan in the FMC, I have on occasion had to switch my runway assignment due to VATSIM controller requests. The original flight plan was imported via SimBrief, and I haven't had issues in the transition from the STAR to the runway approach on those flights where I don't change anything from the original import. HOWEVER - when I have to change the STAR and/or the runway assignment and transition, that's where I get the "Hey, let's go back to the origin airport!" behavior.

Last night, it happened when I was flying KDEN -> KLAX. I was arriving via the ANJELL4 arrival, whose final waypoint is CRCUS. I had had to switch my initial runway assignment to ILS 25L for traffic; when I reached CRCUS, I got the bug. It didn't get fixed when I went DIR to the first waypoint on the runway approach, either - I had to skip to the next waypoint to reestablish my approach.

I'm guessing this is an Asobo issue (as evidenced by the fact that FBW is working on their own custom system for the future). Flying the FBW is rewarding enough that I'm willing to keep getting yelled at by VATSIM for deviations, but I'll also say I'm looking forward to a future without the yelling. :D

**Comment by luiscurtolo** [comment]
> To update everyone, this is something that will be permanently resoled with our custom flight plan system and FMGC, currently being worked on. Stay tuned.

I will start to ask FWB to solve problems of the sim itself as it look like you all handle it better than ASOBO.

You guy are awesome.
Thanks.

**Comment by 2hwk** [comment]
Fixed by 4953 closing ☑️
